test/e2e/tests/confirmations/transactions/metrics.spec.ts

```diff
@@ -1,64 +1,37 @@
 /* eslint-disable @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires */
 import { strict as assert } from 'assert';
-import { MockedEndpoint, MockttpServer } from 'mockttp';
+import { MockttpServer } from 'mockttp';
 import {
   AnonymousTransactionMetaMetricsEvent,
   TransactionMetaMetricsEvent,
 } from '../../../../../shared/constants/transaction';
-import { Driver } from '../../../webdriver/driver';
+import { SMART_CONTRACTS } from '../../../seeder/smart-contracts';
+import { withRedesignConfirmationFixtures } from '../helpers';
 import {
-  confirmContractDeploymentTransaction,
   confirmDepositTransaction,
-  createContractDeploymentTransaction,
   createDepositTransaction,
+  openDAppWithContract,
+  TestSuiteArguments,
 } from './shared';
 
-const {
-  defaultGanacheOptionsForType2Transactions,
-  openDapp,
-  unlockWallet,
-  WINDOW_TITLES,
-  withFixtures,
-  getEventPayloads,
-} = require('../../../helpers');
-const FixtureBuilder = require('../../../fixture-builder');
+const { WINDOW_TITLES, getEventPayloads } = require('../../../helpers');
 
 describe('Metrics', function () {
   if (!process.env.ENABLE_CONFIRMATION_REDESIGN) {
     return;
   }
 
   it('Sends a contract interaction type 2 transaction (EIP1559) with the right properties in the metric events', async function () {
-    await withFixtures(
-      {
-        dapp: true,
-        fixtures: new FixtureBuilder()
-          .withPermissionControllerConnectedToTestDapp()
-          .withPreferencesController({
-            preferences: { redesignedConfirmationsEnabled: true },
-          })
-          .withMetaMetricsController({
-            metaMetricsId: 'fake-metrics-id',
-            participateInMetaMetrics: true,
-          })
-          .build(),
-        ganacheOptions: defaultGanacheOptionsForType2Transactions,
-        title: this.test?.fullTitle(),
-        testSpecificMock: mocks,
-      },
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
       async ({
         driver,
+        contractRegistry,
         mockedEndpoint: mockedEndpoints,
-      }: {
-        driver: Driver;
-        mockedEndpoint: MockedEndpoint;
-      }) => {
-        await unlockWallet(driver);
-
-        await openDapp(driver);
+      }: TestSuiteArguments) => {
+        const smartContract = SMART_CONTRACTS.PIGGYBANK;
 
-        await createContractDeploymentTransaction(driver);
-        await confirmContractDeploymentTransaction(driver);
+        await openDAppWithContract(driver, contractRegistry, smartContract);
```

Comment:
nit: this function is a bit miss-leading, it is called `openDappWithContract` but what it's actually doing is:
- unlocking MetaMask
- fetching the address of the contract
- going to the test dapp with the contract address
Since it's not introduced by this PR, it's not a blocker, but it could be improved in a subsequent PR